File: java/src/test/registry/bucketRegistryTest.java
package test.registry;

import main.registry.bucketRegistry;
import main.time.systemTimeProvider;

public class bucketRegistryTest {
    public static void main(String[] args) {
        // tests for registry

        systemTimeProvider time = new systemTimeProvider();
        bucketRegistry br = new bucketRegistry(20, 6000, time);

        // bucket creation

        br.getBucket("user123");

        // shows buckets that are stored
        
        br.showBuckets();
    }
}

File: typescript/test/registry/BucketRegistry.test.ts
/* Test that perform bucket usage and cleanup */

import { BucketRegistry } from "../../src/registry/BucketRegistry";
import { SystemTimeProvider } from "../../src/time/SystemTimeProvider";

async function run() {
    const time = new SystemTimeProvider();
    const registry = new BucketRegistry(3, 3500, time);

    const b1 = registry.getBucket("user123");
    const b2 = registry.getBucket("user123");

    console.assert(b1 == b2, "Same key must return same bucket");

    const b3 = registry.getBucket("user12");
    console.assert(b1 !== b3, "For different key must return different bucket");

    console.log("Waiting for bucket TTL...");

    await new Promise(r => setTimeout(r, 10000));

    registry.cleanupUnusedBuckets();

    const b4 = registry.getBucket("user123");
    console.assert(b4 !== b1, "Bucket must be recreated after cleanup");

    console.log("BUCKET REGISTRY TEST PASSED");
}